### PR TITLE
Update data.py

### PR DESCRIPTION
**Pull request by drozzy**
Improve internal documentation of the `hoomd.data` module by making implicit variables explicit, and filling in implicit imports. Outcome of the discussion in #453.

### CODE REVIEW

**Comment by joaander** [comment]
@drozzy Thanks. Can you please answer the following questions that the PR template should have provided:
`````
## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->

## Change log

<!-- Propose a change log entry. -->
```

```

## Checklist:

- [ ] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [ ] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [ ] My name is on the [list of contributors](https://github.com/glotzerlab/hoomd-blue/blob/master/sphinx-doc/credits.rst).
`````

**Comment by drozzy** [comment]
## How has this been tested?

No tests required. This is docs only.

## Change log
Improve internal documentation of the hoomd.data module by making implicit variables explicit, and filling in implicit imports.

File: hoomd/data.py [hunk 1]
```diff
@@ -31,28 +31,32 @@
 
 Examples::
 
+    import hoomd
+    system = hoomd.init.create_lattice(unitcell=hoomd.lattice.sc(a=2.0), n=5);
```
Comment (mphoward) [inline hoomd/data.py]: Please remove trailing semicolons throughout, these are not needed in python examples.

File: hoomd/data.py [hunk 2]
```diff
@@ -31,28 +31,32 @@
...
 .. rubric:: Snapshot and MPI
 
 In MPI simulations, the snapshot is only valid on rank 0 by default. make_snapshot, read_snapshot, and take_snapshot,
 restore_snapshot are collective calls, and need to be called on all ranks. But only rank 0 can access data
 in the snapshot::
 
+    import hoomd
+    system = hoomd.init.create_lattice(unitcell=hoomd.lattice.sc(a=2.0), n=5);
     snapshot = system.take_snapshot(all=True)
     if comm.get_rank() == 0:
```
Comment (mphoward) [inline hoomd/data.py]: Please replace `comm.get_rank()` with `hoomd.comm.get_rank()` in all examples you modified

File: hoomd/data.py [hunk 3]
```diff
@@ -782,7 +786,9 @@ def take_snapshot(self,
         in the snapshot
 
         Examples::
-
+            
+            import hoomd
+            system = hoomd.init.create_lattice(unitcell=hoomd.lattice.sc(a=2.0), n=5);
             snapshot = system.take_snapshot()
```
Comment (mphoward) [inline hoomd/data.py]: Please remove this duplicated `take_snapshot` (since you are already cleaning up the docs!).

**Comment by joaander** [comment]
@drozzy Thanks for the changes. I do need you to agree to the contributor agreement before I merge. @mphoward also identified a number of necessary improvements.

**Comment by drozzy** [comment]
... how do I "agree" to it? I don't see any buttons.

**Comment by joaander** [comment]
> ... how do I "agree" to it? I don't see any buttons.

You post this in a comment and put an x in the box.
```
## Checklist:

- [ ] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [ ] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [ ] My name is on the [list of contributors](https://github.com/glotzerlab/hoomd-blue/blob/master/sphinx-doc/credits.rst).
```
You are welcome to add yourself to `credits.rst` or not as you choose, but for licensing reasons I do need a comment that shows you agree to the contributor agreement before I can merge.

**Comment by drozzy** [comment]
[x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
[x ] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).

**Comment by drozzy** [comment]
> @drozzy Do you have any plans to implement @mphoward's requested changes? If not, no worries, this documentation will all be completely rewritten for the v3 API.

Sorry I don't have time now.

**Pull request closed without merging**